vocabularies/Common.json

```diff
@@ -1288,6 +1288,132 @@
+        "$Parameter": [
+          {
+            "$Name": "Entity",
+            "$Type": "Edm.EntityType",
+            "@Core.Description": "See parameter `Entity` in [`Template_RecommendationsFunction`](#Template_RecommendationsFunction)"
+          },
+          {
+            "$Name": "Targets",
+            "$Collection": true,
+            "@Core.IsURL": true,
+            "@Core.Description": "See parameter `Targets` in [`Template_RecommendationsFunction`](#Template_RecommendationsFunction)"
+          }
+        ],
+        "$ReturnType": {
+          "$Collection": true,
+          "$Type": "Common.PropertyRecommendationType",
+          "@Core.Description": "See return type in [`Template_RecommendationsRetrievalFunction`](#Template_RecommendationsRetrievalFunction)"
+        }
+      }
+    ],
+    "PropertyRecommendationType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "target": {
+        "@Core.IsURL": true,
+        "@Core.Description": "URL (relative to the URL of the recommendations function request) addressing a property [OData-URL, section 4.6] that is targeted by this recommendation",
+        "@Core.LongDescription": "The request `SalesOrder('A')/ns.RecommendationsFunction?Targets=...`\n            might retrieve a recommendation with `\"target\": \"Items(10)/Product\"`."
+      },
+      "value": { "@Core.Description": "Recommended value, converted to string" },
+      "text": { "$Nullable": true, "@Core.Description": "Description of the recommended value" },
+      "alternatives": {
+        "$Collection": true,
+        "$Type": "Common.AlternativeRecommendationType",
+        "@Core.Description": "A list of alternative values, sorted by confidence score in descending order",
+        "@Core.LongDescription": "If a value is recommended via property `value`, it must be the first entry in this list."
+      }
+    },
+    "AlternativeRecommendationType": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "value": { "@Core.Description": "Alternatively recommended value, converted to string" },
+      "score": {
+        "$Type": "Edm.Decimal",
+        "$Nullable": true,
+        "@Core.Description": "Confidence score of the alternatively recommended value"
+      }
+    },
+    "RecommendationsRole": {
+      "$Kind": "Term",
+      "$Type": "Common.RecommendationsRoleType",
+      "$AppliesTo": ["Property", "Parameter"],
+      "@Common.Experimental": true,
+      "@Core.Description": "Role of this property or parameter regarding AI-based recommendations"
+    },
+    "RecommendationsRoleType": {
+      "$Kind": "EnumType",
+      "$IsFlags": true,
+      "@Common.Experimental": true,
+      "Input": 1,
+      "Input@Core.Description": "The property is input for some recommendation",
+      "Output": 2,
+      "Output@Core.Description": "The property or parameter is the target of some recommendation",
```

Comment:
We can also have a non-mandatory property for recommendation call marked as input & output role. In runtime if the value for this property is passed it will be used as input param (known value) else AI service may predict the value (will be treated as unknown/output param). Therefore we need to have "InputOutput" also as an enum value

Reply:
Instead of a separate `InputOutput` value, you can give both flags:
```
<Annotation Term="Common.RecommendationsRole"
  EnumMember="Common.RecommendationsRoleType/Input
              Common.RecommendationsRoleType/Output"/>
```

Reply:
Considering that so far we had InOut enums (see valueList) isn't that linked to framework limitation on RAP/CAP side that this cannot be annotated with multiple values and as such we should avoid it ?